flagutil/flagutil.go

```diff
@@ -0,0 +1,73 @@
+// Package containing flag utilities. This includes implementations of flag.Value and flag.Getter interfaces. Also
+// contains methods like ValidateFlags.
+package flagutil
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"strings"
+)
+
+// Validates flags and logs a fatal message with all errors if found. Must be called after flag.Parse().
+// This is done by calling Get() on all flag values which implement the flag.Getter interface and collecting errors if returned.
+func ValidateFlags() {
+	var flagErrors []error
+	checkGetError := func(f *flag.Flag) {
+		if fv, ok := f.Value.(flag.Getter); ok {
+			if err, ok := fv.Get().(error); ok && err != nil {
+				flagErrors = append(flagErrors, flagError(fmt.Sprintf("Flag: %s, Error: %s, Usage: %s", f.Name, err.Error(), f.Usage)))
+			}
+		}
+	}
+	flag.VisitAll(checkGetError)
+	if len(flagErrors) > 0 {
+		errorstr := []string{"Errors found while validating flags"}
+		for _, err := range flagErrors {
+			errorstr = append(errorstr, err.Error())
+		}
+		log.Fatal(strings.Join(errorstr, "\n"))
+	}
+}
+
+type flagError string
+
+func (err flagError) Error() string {
+	return string(err)
+}
```

Comment:
I'm moving my comment from before to here. 

It might be clearer to make `flagError` a struct:
```go
type FlagError struct {
        flag *flag.Flag, // not sure what the best name for this field would be
        err error  // same here
}
```
(I'd export it too, seems like it might be something other packages would want to use)

then you can have:

```go
func (f *flagError) Error() string {
        return fmt.Sprintf("Flag: %s, Error: %s, Usage: %s", f.flag.Name, f.err.Error(), f.flag.Usage)
}
```

What do you think?